playbooks/inventory/dynamic_inventory.py

```diff
@@ -4,10 +4,34 @@
 import subprocess
 import yaml
 
-basedir = os.path.abspath(os.path.dirname(__file__))
-vars_yaml = os.path.join(basedir, '../group_vars/all.yml')
-rpc_repo_path = yaml.load(open(vars_yaml, 'r'))['rpc_repo_path']
 
+def get_rpc_repo_path():
+    """
+    Return the 'rpc_repo_path' configuration setting.
+    The function looks for this setting in the following config files:
+        1. /etc/openstack_deploy/user_extras_variables.yml
+        2. $HOME/openstack_deploy/user_extras_variables.yml
+        3. rpc_extras/playbooks/group_vars/all.yml
+    If the setting isn't found on any of these files then None is returned.
+    """
+    basedir = os.path.abspath(os.path.dirname(__file__))
+    file_check = [
+        '/etc/openstack_deploy/user_extras_variables.yml',
+        os.path.join(os.environ.get('HOME'),
+                     'openstack_deploy/user_extras_variables.yml'),
+        os.path.join(basedir, '../group_vars/all.yml')
+    ]
+
+    for f in file_check:
+        if os.path.isfile(f):
+            cfg = yaml.load(open(f, 'r'))
```

Comment:
NIT: This does not close the file upon load. We should either use "with" or a "try/finally" to ensure that the file descriptor is closed after its been read.